### PR TITLE
[IMP] Payment providers: advertise and document the "demo" provider

### PR DESCRIPTION
**Pull request by toaa-odoo**
New page to document the new features of the demo payment provider, and modifications to related pages

task id: 2982367

### CODE REVIEW

**Comment by toaa-odoo** [comment]
Converted to draft because I want your feedback first, since this is a brand new page, and one feature is to come still (currencies).

**Review by AntoineVDV — COMMENTED** [review]
Short feedback after reading in diagonal:
- The customer fees feature should be documented in `payment_providers.rst` since it's also supported by Alipay and PayPal.
  In fact, the "Demo payment provider" page should not re-explain generic configuration steps.
- I find it cool to talk about the demo provider on the main page, but I'm not sure an entire section is needed for that. Maybe a well-placed tip would be enough?
- `demo.rst` instead of `demo_payment_provider.rst`.

File: content/applications/finance/payment_providers.rst [hunk 1]
```diff
@@ -150,6 +151,13 @@ Online payment providers
    added and configured on your Accounting app to do a bank reconciliation, which is an accounting
    control process.
 
+.. tip::
+   In addition to the regular payment providers that integrate with an API such as Stripe, PayPal,
+   or Adyen, Odoo bundles the :doc:`module <../../../applications/general/apps_modules>`
+   **payment_demo** and the :doc:`demo payment provider <payment_providers/demo>`. This payment
+   provider allows making demo payments to test business flows involving online payments. No
```
Comment (Donapi) [inline content/applications/finance/payment_providers.rst:158]:
```suggestion
   provider allows demo payments to test business flows involving online payments. No
```

File: content/applications/finance/payment_providers/demo.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+~~~~~~~~~~~~~~~~~
+
+You can change the state of a transaction straight from its view form.
+
+.. image:: demo/demo-view-form.png
+   :align: center
+   :alt: Transaction's status view form.
+
+To access a transaction's view form, activate the :ref:`developer mode <developer-mode>`, press
+**CTRL/⌘ + K** and search for `/transactions`. Click either on
```
[the remark this answers is not in the document]
Comment (AntoineVDV) [reply content/applications/finance/payment_providers/demo.rst:46]: I agree. ctrl-k is cool but I find it weird to use it as a shortcut for the full menu path.

[the remark this answers is not in the document]
Comment (AntoineVDV) [reply content/applications/finance/payment_providers/demo.rst:46]: Note that we generally start from the point where the user enabled the provider from the generic "Payment provider" doc, and then give the extra steps.

File: content/applications/finance/payment_providers/demo.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+the desired outcome.
+
+.. image:: demo/demo-payment-outcome.png
+   :align: center
+   :alt: Payment status outcomes.
+
+Transaction state
+~~~~~~~~~~~~~~~~~
+
+You can change the state of a transaction straight from its view form.
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/finance/payment_providers/demo.rst:39]: Changed it up

File: content/applications/finance/payment_providers/demo.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/finance/payment_providers/demo.rst:46]: Added as "tip" instead

**Review by AntoineVDV — COMMENTED** [review]
I already post my pending comments because @jcs-odoo wanted to have a say before the technical review.

**Comment by toaa-odoo** [comment]
Accommodated for changes from https://github.com/odoo/documentation/pull/2775#discussion_r1036132226

File: content/applications/finance/payment_providers/demo.rst [hunk 4]
```diff
@@ -0,0 +1,129 @@
+=====================
+Demo payment provider
+=====================
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:3]:
```suggestion
====
Demo
====
```

"payment provider" is in the category. We wouldn't repeat "payment provider" for all pages in this category. "demo" is no exception.

File: content/applications/finance/payment_providers/demo.rst [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+Odoo's **Demo Payment Provider** allows you to test business flows involving online transactions
+without requiring real banking credentials.
+
+.. seealso::
+   :doc:`../payment_providers`
+
+Installation
+------------
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:12]: Even if it works, we made all headings markups consistent in this repository, please use the same ones as described here
https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#headings

File: content/applications/finance/payment_providers/demo.rst [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+.. image:: demo/demo-payment-provider.png
+   :align: center
+   :alt: Demo Payment Provider icon.
+
+Configuration
+-------------
+
+The **Demo Payment Provider** has multiple configuration options that can be accessed by going to
+:menuselection:`Accounting or Website app --> Configuration --> Payment Providers`.
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:26]: See this review's comment for more explanation about this.
Also, even if it works, we made all headings markups consistent in this repository, please use the same ones as described here
https://www.odoo.com/documentation/master/contributing/documentation/rst_cheat_sheet.html#headings
```suggestion
Configuration
=============

.. seealso::
   :ref:`payment_providers/add_new`

.. important::
   switch its state to :guilabel:`Test Mode`.
```

File: content/applications/finance/payment_providers/paypal.rst [hunk 7]
```diff
@@ -38,14 +38,14 @@ token by following the configuration step :ref:`paypal/enable-pdt`.
...
------------------
+Fees tab
+--------
 
 You can charge extra fees to your customers for paying with Paypal to cover the transaction fees
 Paypal charges you. Once redirected to Paypal, your customer sees an extra amount applied to the
 order amount.
 
-To activate this, go to Paypal configuration's Configuration tab in Odoo and activate *Add Extra
+To activate this, go to Paypal configuration's :guilabel:`Fees` tab in Odoo and activate *Add Extra
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/paypal.rst:48]: This should be in a different commit too.
This change is valid from 14.0

**Review by jcs-odoo — CHANGES_REQUESTED** [review]
Hello
@toaa-odoo @xpl-odoo  (Please read my feedback carefully too, Xavier =) )

The last payment providers added to this category or updated seemed to have a new structure, which I think is quite good. I assume that you agreed with it since it's been merged.
(e.g., [Adyen](https://www.odoo.com/documentation/16.0/applications/finance/payment_providers/adyen.html), [Authorize.Net](https://www.odoo.com/documentation/16.0/applications/finance/payment_providers/authorize.html), [PayPal](https://www.odoo.com/documentation/16.0/applications/finance/payment_providers/paypal.html) etc.)

For consistency, it's best to keep that same structure.

1. Configuration
   1. Credentials tab (not useful here)
   2. Configuration tab 
2. (specific to this demo payment provider) --> how to use it (payment outcome etc)

In these pages, custom anchor links are added whenever possible rather than reexplaining what's described already in the main payment providers page (for example, how to add a new payment provider) 

Note that on these pages, we only reference the settings that need further explanation specific to the payment provider at hand.

A lot of this page's content isn't specific to this payment provider and should be dropped as it is already explained on the main page as well (e.g., displayed as, supported payment icons, etc.). If it isn't well explained on that page, you can work on improving that content in a separate task.

In the end, this page will be short, and that will be better (easier to understand and less maintenance)

File: content/applications/finance/payment_providers/paypal.rst (hunk 7 above)
Comment (toaa-odoo) [reply content/applications/finance/payment_providers/paypal.rst:48]: https://github.com/odoo/documentation/pull/3074

File: content/applications/finance/payment_providers.rst [hunk 8]
```diff
@@ -63,83 +64,96 @@ Online payment providers
    :widths: auto
 
    * -
-     - Payment flow
+     - Payment flow (from)
```
[the remark this answers is not in the document]
Comment (toaa-odoo) [reply content/applications/finance/payment_providers.rst]: https://github.com/odoo/documentation/pull/3075

File: content/applications/finance/payment_providers/demo.rst [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+Configuration --> Payment Transactions`. Then, change the status of a transaction by clicking on the
+state bar (:guilabel:`Draft, Pending, Authorized, Confirmed, Canceled, Error`).
+
+.. image:: demo/demo-view-form.png
+   :align: center
+   :alt: Transaction's status bar.
+
+.. tip::
+   You can also access the **transactions menu** by pressing **CTRL/⌘ + K** and searching for
+   `/transactions`.
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:43]: this tip is true, but we could add it pretty much everywhere in the doc, no?

File: content/applications/finance/payment_providers/demo.rst [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+   the card details directly. Instead, it creates a payment token that only holds a reference to the
+   card details stored on the payment provider’s server.
+
+Manual capture
+--------------
+
+Enable :guilabel:`Capture Amount Manually` to test payment captures in two steps instead of one.
+When you authorize a payment, the funds are (in the case of real transactions) reserved on the
+customer's payment method but they are not immediately charged. The charge is only made when you
+manually decide to capture the payment at a later date.
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:63]: I don't understand why you left these two sections, after my last review?

File: content/applications/finance/payment_providers/demo.rst [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+====
+
+Odoo's **Demo Payment Provider** allows you to test business flows involving online transactions
+without requiring real banking credentials.
+
+Configuration
+=============
+
+.. seealso::
+   - :ref:`payment_providers/add_new`
```
Comment (jcs-odoo) [inline content/applications/finance/payment_providers/demo.rst:12]:
```suggestion
.. seealso::
   - :ref:`payment_providers/add_new`
   
   .. important::
   Switch the state to :guilabel:`Test Mode`.
```

**Review by jcs-odoo — CHANGES_REQUESTED** [review]
- don't forget to delete the images that you don't use.

I'll apply all the changes myself.

**Comment by jcs-odoo** [comment]
@robodoo r+
